src/pages/Profile.js

```diff
@@ -1,87 +1,258 @@
 import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
 import { ChartContainer } from '@codeforafrica/hurumap-ui';
+import gql from 'graphql-tag';
+import { useQuery, useApolloClient } from 'react-apollo-hooks';
+import { Grid } from '@material-ui/core';
 import { ProfilePageHeader } from '../components/Header';
 import ProfileTabs from '../components/ProfileTabs';
 import Page from '../components/Page';
 import CountryPartners from '../components/CountryPartners';
 import config from '../config';
-import { getProfile, getComparisonProfile } from '../lib/api';
 import ChartFactory from '../components/ChartFactory';
 import ChartsContainer from '../components/ChartsContainer';
 
+import sectionedCharts from '../data/charts.json';
+
 function Profile({
   match: {
-    params: { geoId, anotherGeoId }
+    params: { geoId, comparisonGeoId }
   }
 }) {
-  const head2head = Boolean(geoId && anotherGeoId);
-  const [activeTab, setActiveTab] = useState('');
-  const [selectedCountry, setSelectedCountry] = useState({});
-  const [profile, setProfile] = useState();
+  const head2head = Boolean(geoId && comparisonGeoId);
+  const [activeTab, setActiveTab] = useState('All');
+  const [profiles, setProfiles] = useState({
+    profile: {},
+    parentProfile: {},
+    comparisonProfile: {}
+  });
+  const client = useApolloClient();
+
+  // Provide the visuals with unique ids for fetching
+  // The unique ids will be used to set alias in graphql
+  let index = 0;
+  sectionedCharts.forEach(x =>
+    x.charts.forEach(y =>
+      y.visuals.forEach(z => {
+        // eslint-disable-next-line no-param-reassign
+        z.id = `viz${index}`;
+        index += 1;
+      })
+    )
+  );
+
+  // Flatten all charts
+  const charts = sectionedCharts
+    .map(x => x.charts)
+    .reduce((a, b) => a.concat(b));
+  const visuals = charts.map(x => x.visuals).reduce((a, b) => a.concat(b));
+
+  const sections = ['All', ...sectionedCharts.map(x => x.sectionTitle)];
+
+  // Build chart data query
+  const visualsQuery = gql`
+  query charts($geoCode: String!, $geoLevel: String!) {
+    ${visuals
+      .map(
+        visual => `${visual.id}: ${visual.table} (
+      condition: { geoCode: $geoCode, geoLevel: $geoLevel }
+    ) {
+      nodes {
+        ${
+          visual.label && visual.label[0] === '$'
+            ? `label: ${visual.label.slice(1)}`
+            : ''
+        }
```

Comment:
I don't understand the dollar sign on label.

From the if statement, If we have a defined label and first letter is not '$', we do not set a label at all. Is this correct?

Reply:
So if we specify it with a dollar sign. I take it as a variable we go to the db to fetch. Example is the "$longName". I left it there as an example. We can fetch the geo name by just doing that. If we had more relationships, this label could be made to fetch the parent or profile geo name for example, without us having to prefetch the parent geo and/or profile geo.

So if there is no dollar sign i assume you want to query `longName` from the prefetched profiles. If the label does not exist in that profile then I assume the lable is just a string you want to use e.g. `label: "People"`

@KhadijaMahanga

Reply:
I know its a bit of a circle but since we want default values, we have it working like this.